earth_enterprise/src/common/khTypes.h

```diff
@@ -62,6 +62,61 @@ typedef double                          float64;
 
 #ifdef __cplusplus
 
+// used to limit the number of strings in memory
+// thread-safe and removes when no longer needed
+#include <boost/flyweight.hpp>
+
+class AssetHandle
+{
+    boost::flyweights::flyweight<std::string> fw;
+public:
+    AssetHandle() = default;
+    AssetHandle(const AssetHandle& other) { fw = other.fw; }
```

Comment:
It looks like a lot of these functions (copy constructor, copy assignment operator, etc) are just the default versions and can be removed.